clippy_lints/src/return_self_not_must_use.rs

```diff
@@ -13,25 +14,46 @@ declare_clippy_lint! {
     /// This lint warns when a method returning `Self` doesn't have the `#[must_use]` attribute.
     ///
     /// ### Why is this bad?
-    /// It prevents to "forget" to use the newly created value.
+    /// Methods returning `Self` often create new values, having the `#[must_use]` attribute prevents
+    /// users from "forgetting" to use the newly created value.
+    ///
+    /// The `#[must_use]` attribute can be added to the struct or enum itself to ensure that instances
```

Comment:
I would write type instead of struct or enum. It's shorter and also includes unions.